zerver/actions/streams.py

```diff
@@ -1672,6 +1676,27 @@ def do_change_stream_group_based_setting(
         name=stream.name,
     )
     send_event_on_commit(stream.realm, event, current_user_ids_with_metadata_access)
+    if setting_name in Stream.stream_permission_group_settings_granting_metadata_access:
+        user_ids_gaining_metadata_access = (
+            current_user_ids_with_metadata_access - old_user_ids_with_metadata_access
+        )
+        user_ids_losing_metadata_access = (
+            old_user_ids_with_metadata_access - current_user_ids_with_metadata_access
+        )
+
+        recent_traffic = get_streams_traffic({stream.id}, stream.realm)
+        setting_groups_dict = get_group_setting_value_dict_for_streams([stream])
```

Comment:
We should only this work if we need it; these lines can go inside the first `if` statement.